bindings/wasm/native/src/client_builder.rs

```diff
@@ -0,0 +1,265 @@
+    let mut clientbuilder = self.clone();
+    let promise: Promise = future_to_promise(async move {
+      clientbuilder.builder = Some(
+        clientbuilder
+          .take_builder()?
+          .with_node_pool_urls(&node_pool_urls[..])
+          .await
+          .map_err(wasm_error)?,
+      );
+      Ok(clientbuilder.into())
+    });
+    Ok(promise)
+  }
+
+  /// Set if quroum should be used or not
+  #[wasm_bindgen(js_name = quorum)]
+  pub fn quorum(&mut self, value: bool) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_quorum(value))?;
+    Ok(self.clone())
+  }
+
+  /// Set amount of nodes which should be used for quorum
+  #[wasm_bindgen(js_name = quorumSize)]
+  pub fn quorum_size(&mut self, value: usize) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_quorum_size(value))?;
+    Ok(self.clone())
+  }
+
+  /// Set quorum_threshold
+  #[wasm_bindgen(js_name = quorumThreshold)]
+  pub fn quorum_threshold(&mut self, value: usize) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_quorum_threshold(value))?;
+    Ok(self.clone())
+  }
+
+  /// Selects the type of network to get default nodes for it, only "testnet" is supported at the moment.
+  /// Nodes that don't belong to this network are ignored. Default nodes are only used when no other nodes are
+  /// provided.
+  #[wasm_bindgen]
+  pub fn network(mut self, network: &str) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_network(network))?;
+    Ok(self)
+  }
+
+  /// Since we can only have a single thread in wasm, local PoW is much slower
+  #[wasm_bindgen(js_name = localPow)]
+  pub fn local_pow(&mut self, value: bool) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_local_pow(value))?;
+    Ok(self.clone())
+  }
+
+  /// Sets after how many seconds new tips will be requested during PoW
+  #[wasm_bindgen(js_name = tipsInterval)]
+  pub fn tips_interval(&mut self, value: u32) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_tips_interval(u64::from(value)))?;
+    Ok(self.clone())
+  }
+
+  /// Sets the default request timeout.
+  #[wasm_bindgen(js_name = requestTimeout)]
+  pub fn request_timeout(&mut self, value: u32) -> Result<ClientBuilder, JsValue> {
+    self.with_mut(|builder| builder.with_request_timeout(to_duration(value)))?;
+    Ok(self.clone())
+  }
+
+  /// Sets the request timeout for a specific API usage.
+  #[wasm_bindgen(js_name = apiTimeout)]
+  pub fn api_timeout(&mut self, api: String, timeout: u32) -> Result<ClientBuilder, JsValue> {
+    let api = Api::from_str(&api)
+      .map_err(|_| iota_client::Error::ApiError)
+      .map_err(wasm_error)?;
+    self.with_mut(|builder| builder.with_api_timeout(api.clone(), to_duration(timeout)))?;
+    Ok(self.clone())
+  }
+
+  pub(crate) fn take_builder(&mut self) -> Result<RustClientBuilder, JsValue> {
```

Comment:
Whats up with all this shenanigans? This works fine :)
https://github.com/kwek20/iota.rs/commit/b5be0590dd11e33d256b8fa9829cbcfc0c623ee1

Or is it for some error reason, you catch and can continue the builder? Im not sure if thats really needed, if your builder fails, your program likely does as well

Reply:
Thanks for pointing this out, yes I think your approach is better, will merge it